dev-libs/ferrisloki/files/ferrisloki-3.0.13-r3-configure-libsigc.patch

```diff
@@ -0,0 +1,66 @@
+Rollup of src_prepare() from previous revisions plus additional fixes
+
+ * derice damn configure script
+ * fix building with libsgc++-2.6
+ * fix building with /bin/sh -> dash
+
+--- a/configure
++++ b/configure
+@@ -15228,7 +15228,7 @@ have_stlport=no
+ if test "${with_stlport+set}" = set; then :
+   withval=$with_stlport; STLPORT_CFLAGS=" -I$withval/stlport "
+ 	 STLPORT_LDFLAGS=" -L$withval/lib "
+- 	 STLPORT_LIB=" -lstlport_gcc -lpthread "
++ 	 STLPORT_LIB=" -lstlport -lpthread "
+ 	 STLPORT_LIBS=" -L$withval/lib ${STLPORT_LIB} "
+ 	 stlport_try_trivial_compile=yes
+
+@@ -15540,10 +15540,10 @@ fi
+ 		stlport_installed_version=`stlport-config --version`
+
+ 		# Calculate the available version number
+-		f_tmp=( `echo $stlport_installed_version | sed 's/[^0-9]\+/ /g'` )
++		f_tmp=" `echo $stlport_installed_version | sed 's/[^0-9]\+/ /g'` "
+ 		f_tmp=$(( 1000000 * ${f_tmp[0]:-0} + 1000 * ${f_tmp[1]:-0} + ${f_tmp[2]:-0} ))
+
+-		freq_version=( `echo $stlport_required_version | sed 's/[^0-9]\+/ /g'` )
++		freq_version=" `echo $stlport_required_version | sed 's/[^0-9]\+/ /g'` "
```

Comment:
These two lines are the only ones I changed to make the build go through.

Reply:
great finding and fix 😄

Reply:
It's been pointed out to me that it's probably intentional that f_tmp is created as an array, so I probably need to figure out how to do that in dash.

Reply:
So: This part of the configure script I've patched is only used when the stlport feature is enabled _and_ it wants to run the `stlport-config --version` command to identify the major/minor/patch version numbers (at least that's how I'm understanding the code at the moment).

But the ebuild explicitly `econf --disable-stlport` so that feature is never enabled.

The goal was to make the configure script work with dash (which arguably it does), but _technically_ it's broken because the variables `f_tmp` and `freq_version` are supposed to be bash arrays and not just strings.

What's the preferred fix here? Completely remove the section of the configure script? Or try to make it work just in case stlport is ever re-enabled?

Reply:
Update: stlport has been dead for years and was last-rited [some time ago](https://archives.gentoo.org/gentoo-dev/message/ff7cfa496b43c2c098021eee255891a8).

I've patched configure.in now to remove the probing for stlport-config, and calling eautoreconf.